homeassistant/components/trend/binary_sensor.py

```diff
@@ -141,12 +144,10 @@ def __init__(
         self._invert = invert
         self._sample_duration = sample_duration
         self._min_gradient = min_gradient
-        self._gradient = None
-        self._state = None
-        self.samples = deque(maxlen=max_samples)
+        self.samples: deque = deque(maxlen=max_samples)
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool | None:
         """Return true if sensor is on."""
         return self._state
 
```

Comment:
extra state attributes does not have a return type

Reply:
Good catch, thanks 👍